x-pack/plugins/session_view/public/components/ProcessTreeNode/index.tsx

```diff
@@ -0,0 +1,313 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ *2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useMemo, useRef, useLayoutEffect, useState, useEffect, MouseEvent } from 'react';
+import { EuiButton, EuiIcon, EuiToolTip } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { Process } from '../../../common/types/process_tree';
+import { useStyles, ButtonType } from './styles';
+import { ProcessTreeAlerts } from '../ProcessTreeAlerts';
+
+interface ProcessDeps {
+  process: Process;
+  isSessionLeader?: boolean;
+  depth?: number;
+  onProcessSelected?: (process: Process) => void;
+}
+
+/**
+ * Renders a node on the process tree
+ * TODO: as well as sections for tty output, alerts and file redirection.
+ */
+export function ProcessTreeNode({
+  process,
+  isSessionLeader = false,
+  depth = 0,
+  onProcessSelected,
+}: ProcessDeps) {
+  const textRef = useRef<HTMLSpanElement>(null);
+
+  const [childrenExpanded, setChildrenExpanded] = useState(isSessionLeader || process.autoExpand);
+  const [alertsExpanded, setAlertsExpanded] = useState(false);
+  const [showGroupLeadersOnly, setShowGroupLeadersOnly] = useState(isSessionLeader);
+  const { searchMatched } = process;
+
+  useEffect(() => {
+    setChildrenExpanded(isSessionLeader || process.autoExpand);
+  }, [isSessionLeader, process.autoExpand]);
+
+  const processDetails = useMemo(() => {
+    return process.getDetails();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [process.events.length]);
+
+  const hasExec = useMemo(() => {
+    return process.hasExec();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [process.events.length]);
+
+  const alerts = useMemo(() => {
+    return process.getAlerts();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [process.events.length]);
+
+  const styles = useStyles({ depth, hasAlerts: !!alerts.length });
+
+  useLayoutEffect(() => {
+    if (searchMatched !== null && textRef.current) {
+      const regex = new RegExp(searchMatched);
+      const text = textRef.current.textContent;
+
+      if (text) {
+        const html = text.replace(regex, (match) => {
+          return `<span data-test-subj="processNodeSearchHighlight" style="${styles.searchHighlight}">${match}</span>`;
```

Comment:
data test subj update: `data-test-subj="xpack.sessionViewer.processNodeSearchHighlight"`

Reply:
@animehart can you address the `data-test-subj` changes?

Reply:
Yep, I will address test related feedback